build.gradle

```diff
@@ -228,6 +260,26 @@ project('reactor-extra') {
 		maven { url "https://maven-eclipse.github.io/maven" }
 	}
 
+	ext {
+		bndOptions = [
+			"Export-Package": [
+				"!*internal*",
+				"reactor.*;version=$osgiVersion;-noimport:=true"
```

Comment:
This one might be tricky. The reactor-extra packages do not start with a common "reactor.extra" package name therefore we would need to explicitly list the packages contained in this module e.g. "reactor.bool", "reactor.cache", "reactor.extra.processor", "reactor.function", "reactor.math", ...

Or we try something like:
-exportcontents: reactor.*;version=$osgiVersion;-noimport:=true
instead of "Export-Package"

As I have stated above -exportcontents only considers the packages in the module itself and not the complete classpath of the module.